### PR TITLE
add test for array constructor

### PR DESCRIPTION
**Pull request by v923z**
This PR adds a simple test to demonstrate the difference between `CPython` and `micropython` for the case, when an array is constructed with values that do not fit the container. 

The corresponding issue can be found under https://github.com/micropython/micropython/issues/7377

### CODE REVIEW

File: tests/cpydiff/module_array_constructor.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+"""
+categories: Modules,array
+description: overflow checking is not implemented
+cause: micropython implements implicit truncation in order to reduce code size and execution time
+workaround: mask value
+"""
+import array
+
+value = 257
+a = array.array("b", [value & 0xff])
```
Comment (dpgeorge) [inline tests/cpydiff/module_array_constructor.py:10]: the idea is for this test to produce different results on CPython vs MicroPython, so you need to remove the `& 0xff`

File: tests/cpydiff/module_array_constructor.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+"""
+categories: Modules,array
+description: overflow checking is not implemented
+cause: micropython implements implicit truncation in order to reduce code size and execution time
+workaround: mask value
```
Comment (dpgeorge) [inline tests/cpydiff/module_array_constructor.py:5]: I suggest: "if CPython compatibility is needed then mask the value explicitly"

**Review by dpgeorge — CHANGES_REQUESTED** [review]

File: tests/cpydiff/module_array_constructor.py (hunk 1 above)
Comment (v923z) [reply tests/cpydiff/module_array_constructor.py]: OK, done.

File: tests/cpydiff/module_array_constructor.py (hunk 2 above)
Comment (v923z) [reply tests/cpydiff/module_array_constructor.py]: I have updated the PR.

**Comment by dpgeorge** [comment]
Thanks for updating, merged in c4ed17ff345ee59f284a2f6d1ec18034eeb311a9

**Pull request closed without merging**